src/detect-engine.c

```diff
@@ -2342,7 +2342,15 @@ static int DetectEngineCtxLoadConf(DetectEngineCtx *de_ctx)
             }
 
             if (insp_recursion_limit != NULL) {
-                de_ctx->inspection_recursion_limit = atoi(insp_recursion_limit);
+                if (ByteExtractStringInt32(&de_ctx->inspection_recursion_limit,
+                    10, 0, (const char *)insp_recursion_limit) < 0) {
+                    SCLogWarning(SC_ERR_INVALID_VALUE, "Invalid value for "
+                            "detect-engine.inspection-recursion-limit, "
+                            "resetting to %d",
+                            DETECT_ENGINE_DEFAULT_INSPECTION_RECURSION_LIMIT);
+                    de_ctx->inspection_recursion_limit =
+                        DETECT_ENGINE_DEFAULT_INSPECTION_RECURSION_LIMIT;
```

Comment:
This causes test 2 of this module to fail. Here, with this change, no matter what `insp_recursion_limit` is, I am always setting it to `DETECT_ENGINE_DEFAULT_INSPECTION_RECURSION_LIMIT`. However, earlier (i.e. in current master), for cases where `inspection-recursion- limit` used to be empty in `suricata.yaml`, the value was set to `0` which led to a further condition check later on (L2353 in master) and resetting the value to `-1` which (from my understanding) is invalid and causes the test to fail as it was expecting `-1`.
Question: Should the test be fixed or the value setting done above?

Reply:
Good question. I'll defer to the others on this one.

Reply:
It looks like the ```-1``` is used to disable it, but that doesn't make sense as a default value given that we are expected to set it to ```DETECT_ENGINE_DEFAULT_INSPECTION_RECURSION_LIMIT```. Please update the tests to match the ```DETECT_ENGINE_DEFAULT_INSPECTION_RECURSION_LIMIT``` value.